src/main/java/cubicchunks/asm/mixin/core/common/MixinWorldServer.java

```diff
@@ -72,16 +75,18 @@
 		this.isCubicWorld = true;
 
 		this.entitySpawner = new CubeWorldEntitySpawner();
-		this.chunkProvider = new ServerCubeCache(this);
-		this.generatorPipeline = new GeneratorPipeline(this);
-		this.thePlayerManager = new PlayerCubeMap(this);
-		this.chunkGc = new ChunkGc(getCubeCache(), getPlayerCubeMap());
+		ServerCubeCache chunkProvider = new ServerCubeCache(this);
+		this.chunkProvider = chunkProvider;
+		this.generatorPipeline = new GeneratorPipeline();
 		this.lightingManager = new LightingManager(this);
 
-
 		ICubicChunksWorldType type = (ICubicChunksWorldType) this.getWorldType();
 		type.registerWorldGen(this, this.generatorPipeline);
-		this.generatorPipeline.checkStages();
+		GeneratorPipeline.checkStages(this.generatorPipeline);
+		this.cubeGenerator = new WorldGenerator(this, this.generatorPipeline);
```

Comment:
If you really want to go by definition, then GeneratorPipeline isn't a single class but a whole package + CubeProcessors.

And as you said, ICubeGenerator doesn't really do the processing, but it is the active part that manages individual cube and it's also the entry point for cubes into the pipeline. So it would make more sense for it to be named GeneratorPipeline.

And for classes other than ICubeGenerator itself, current GeneratorPipeline can be used only to add (or register) new generator stages. And it can be used to retrieve the actual stages using some properties - in this case name. And you can iterate over all stages - this is also obvious functionality for a registry. And you can get the first stage - object with a special property.